### DISCUSSION TITLE
[Feature] Shared Albums in Search Results

### DISCUSSION THREAD

**Discussion by andrewdunndev** [immich-app/immich#2995] [Feature Request]
### Shared Albums in Search Results

With #2900 added it would be nice to include albums shared with the user in the search results.

I'd discussed a bit [here](https://github.com/immich-app/immich/discussions/2361#discussioncomment-5786345), but my use case is that I would rather digest historic images into immich directly (rather than use the ["read-only/existing library"](https://github.com/immich-app/immich/pull/2715)). For my implementation I've got 500+ albums that my administrative user has added to the server and shared to the other users based on relevant participation.

Those users have to scroll through a lot of albums to find what they are looking for, search would be great.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by andrewdunndev** [comment]
I've moved over my family now and have closer to 1000 albums. I'd happily pay a bounty for this to be implemented!

**Reply by martabal** [reply]
I can take a look at this tonight

**Reply by martabal** [reply]
Well, I think I'll wait https://github.com/immich-app/immich/pull/3605 to be merged before working on it

**Reply by jrasm91** [reply]
You could add a dedicated search/filter bar on the album page itself.

**Reply by martabal** [reply]
I started to implement it but I realized that a single search bar for the whole web app is probably the best choice

**Reply by andrewdunndev** [reply]
Would still happily bounty this!

**Reply by goranbrkic** [reply]
> Well, I think I'll wait https://github.com/immich-app/immich/pull/3605 to be merged before working on it

As above PR has been merged, is it possible for this feature request to get some attention again?

**Reply by BenjaminDo** [reply]
That would be a cool feature ! Smart search is great for my usage but would be perfect if it works in all accessible phots

**Reply by greirson** [reply]
This would be great. I have about 1200 physical family photos that we're looking to scan and digitize. It would be amazing if we could share albums with family and have people already tagged for easy searching

**Reply by enfyrneaux** [reply]
I'll also add to the bounty. 

I'm currently evaluating Immich for collaborative comments/categorization of about 40,000 family archive photos that have been scanned in from the past 80 years. Since multiple relatives are working on this project, all scanned photos have been bulk uploaded using a separate admin user - unfortunately the accounts of my relatives cannot leverage the smart search that I chose Immich for in the first place :neutral_face:

**Reply by xamino** [reply]
Would it be worth to split this into two tasks?
1. Search for a shared album: if you have a lot of shared albums at least allows you to jump to one that you know exists (by year, name, etc). Having this in the mobile app would be great because there is no way to find an album without a lot of scrolling (since you can't filter or sort the "Sharing" tab in any way).
2. Search that works across shared content, which some commenters here seem to be looking for, which is different if not even more useful, though likely harder to implement.

**Reply by bentrnr21** [reply]
Very nice good job

**Reply by GhostDog98** [reply]
This issue still seems to be open, which is a shame... I'd also be willing to set a bounty for this issue to be implemented, or help in any way I can.

**Reply by bo0tzz** [reply]
@GhostDog98 We don't really want bounties (see [ziglang.org/news/bounties-damage-open-source-projects](https://ziglang.org/news/bounties-damage-open-source-projects/) for some reasons). Of course, you're welcome to hire someone directly to implement a feature (but please check in beforehand if you're not sure whether we would accept it).

**Reply by GhostDog98** [reply]
While I don't necessarily agree with that point of view myself, I respect that's how you would like to develop this project, so with that I will withdraw the bounty offer :+1:.

**Comment by suukit** [comment]
+1

**Comment by DonRichie** [comment]
My use case is that I want to filter for only videos in a shared album, that I am not the owner of.
It has ~2000 files, which is so much, that I miss the videos when scrolling through.

**Comment by jeanius7** [comment]
I would love this to be a feature to be worked on! I have easily 12,000 pictures all organized by album. Being able to share some of these folders to my family and them being able to search would be a HUGE quality of life improvement to shared albums.

**Comment by haoenz** [comment]
Will this feature be added to the roadmap?

**Comment by miguelbrandao** [comment]
Just chiming in to say this feature would really be appreciated. I use Immich for two main purposes:

- as photo and video manager for myself (Google Photos/iCloud replacement);
- as a simple, centralized service for friends/family members to access and explore our memories.

Immich accomplishes the former better than any other solution I've found, but it's sorely lacking in the latter: because my friends/family members are just viewers of media I upload and manage, with no photos of their own, all their content is in the form of shared albuns.
Currently, this means they can't use the features that truly distinguish Immish:
- the timeline;
- the explore page;
- smart search;
- map;
- people;

And it is a shame, as the experience I have with my administration account (the owner of all files) is so superior to the experience a friend has, that one could mistake it for different software!

I'll definitely add to the bounty if anyone develops this!

**Comment by rmarkus** [comment]
I would also like to have this supported. We share family photos from a central "family" account; these are the by far the most images. To have the search available for each account would really be great. On the other hand we cannot just use "1 account" since the members have additionally their "own" photos that they don't want to share.

**Comment by DevonMark** [comment]
Immich is so far great, but not having search available for shared assets albums is a major showstopper and means it is almost unusable. Use case: Family members have all taken photos on a holiday and have uploaded to own accounts and then shared to a "Summer Holiday" album. Search only finds _your own pictures_  in that album. I can't see any way around it aside from only having one account (as commented above). I am, sadly, looking for other self-hosted software solutions, but hoping that some form of global search might be implemented before I find one!

**Comment by mcooper-gb** [comment]
+1 for this feature request

**Comment by tillepprecht** [comment]
I would really like this to be implemented as well :) It would make immich so perfect for sharing large ammounts of pictures of events, where people could then filter for pictures they are in, or by tags (eg if the event had multiple parts, etc)

**Comment by Jakiezweers** [comment]
Hi!
Are there any plans to add search support for shared albums?
I just started using Immich and would really love to be able to search within shared content — it feels like a natural feature to have.
Thanks for all the great work!

**Comment by mikebakke** [comment]
Immich is almost perfect but I find this topic really confusing together with a related one.

Direct topic first - the lack of image search when an album is shared really is (as multiple others have already stated) a show-stopper for me to adopt. It's great for me to manage collections but practically it's pointless giving others access because photos in any moderate sized libraries are impossible (pretty much) to find - repeated scrolling is unpleasant and a time sink.

I 've tried every work around and the closest I've come to is curate as admin, then create another user, create the target album and import as that user. If I share those credentials then users can search. However (related issue) no admin function can be performed on the album because the owner is non admin and there's no capabilty to grant admin privs to another user. 

The priority really has to be (IMO) to let admin enable search privileges for any photos in a shared album if desired. 

Not shouting for it "now" but would be fantastic if it was at least confirmed as a firm roadmap item.

**Comment by ashok-dangol** [comment]
First of all, thank you for the amazing work on Immich — it’s an outstanding platform!

We’ve been in the process of deploying Immich across our organization, and two feature requests have come up consistently from our end users that would significantly improve usability:

🔍 1. Search Support within Shared Albums
Currently, search functionality works great within personal libraries, but it doesn't seem to support shared albums. Being able to search within shared content (e.g., by filename, tags, or AI-detected objects) would be incredibly useful, especially in collaborative environments where multiple users contribute to shared albums.

Use Case:
Team members collaborating on media-heavy projects (e.g., marketing events, research photos) need to quickly find specific content within a shared album without scrolling through hundreds of items.

📁 2. Folder Support within Albums
We also received requests for the ability to create folders within albums for better organization. As some albums can grow quite large, a simple folder hierarchy would allow users to categorize content (e.g., by event, location, or contributor) and streamline access.

Use Case:
In large shared albums from multi-day events, being able to sort media into folders like “Day 1”, “Behind the Scenes”, or “Keynote” would help users locate content much faster and keep things clean.

🙏 Request
We’d love to know if either of these features is on the roadmap or open for contribution. Both would greatly enhance the user experience in team-oriented deployments.

Thanks again for all your work — Immich is a fantastic project, and we’re excited to support and adopt it across our organization!

**Comment by ihlstbrown** [comment]
+1 for this whole post. People, folders, search... Please! Tell me where to donate. (seriously)

**Comment by Chaoscontrol** [comment]
New user here, getting used to Immich current limitations. 

Just voted for this, but just wanted to clarify that this request is specific for Shared Albums, right? I mean, this works when done via "Partner Sharing", and after enabling "Show in Timeline", you can search for everything as well. 

Maybe it serves as a workaround for some of you.

**Reply by Nuvial** [reply]
This allows searching and filtering features but doesn't translate over to the facial detection which would be useful.

**Comment by orgildinio** [comment]
Bumping this request.

Creating individual albums for each person is too troublesome, and managing a family album with more than 50 uploaders gets nerve-wracking after reaching more than 200,000 images and videos.

I’d like to put a bounty on this one — whoever builds this feature and gets it merged, I’d love to say thanks and send over some coffee money to support you!

@alextran1502 @jrasm91 ❤️

**Reply by bo0tzz** [reply]
You're welcome to engage with someone directly, but we prefer not to have open bounties. The zig project explains it well: https://ziglang.org/news/bounties-damage-open-source-projects/. There's also no need to ping maintainers.

**Reply by orgildinio** [reply]
If I hire a third-party developer to do this, will it be accepted for merging?

Or can someone earn money with immich code base to make custom jobs?

thank you for clarification on bounty thing <3

**Reply by WizenThorne** [reply]
> You're welcome to engage with someone directly, but we prefer not to have open bounties. The zig project explains it well: https://ziglang.org/news/bounties-damage-open-source-projects/. There's also no need to ping maintainers.

I clicked your link with the mindset that I disagreed, because this is a feature I think Immich really does need and it would benefit me personally. It's also something I would be willing to contribute money towards.

However, as I was reading the article I came to the realization that bounties in open source projects are no different from how people with more money in everyday life have access to getting their own way and better EVERYTHING, from medical care, housing options, and countless other basic needs of life, not to mention all the stuff we as humans don't need. It even dictates whether or not someone will be treated justly in the US criminal court system and the severity of traffic violations.

Unfortunately a lot of people are unwilling to change their stance on something once they've made up their minds, but I'm not one of those people. I wholeheartedly agree that Immich and other open source projects need to be free from the influence of money and the power someone with MORE money can leverage against those with less.

**Reply by CybranM** [reply]
I agree with the sentiment around bounties. I do really want the feature though. Tricky situation

**Comment by killervette42** [comment]
I would like this feature too.

**Comment by tech-guru-49** [comment]
I onboarded my family and uploaded 8,000+ items into a shared album before realizing they won't be able to search the album. :( The ability to search shared albums is really important to us, and we hope it can be added as a feature.

**Comment by Fantasdreas** [comment]
I would really appreciate the addition of this feature. Since sharing albums and photos is one of the main reasons I use Immich, it would be a significant improvement for me. Otherwise, great work so far—I love what you’ve built!

**Comment by DJF3** [comment]
Unfortunately I just came to the same conclusion. One of the key Immich features is the fantastic search, face recognition and maps. **_SHARED_ folders are not included in**
- search
- face recognition
- maps

It would be so fantastic if these features would be included when sharing an album with users. Why keep the great experience from the people you share photos with?

I would even go as far as saying that this is the last thing blocking me from deploying across friends & family. (and buying a license)
Why keep these great features from the people you share photos with?

**Comment by Evigami** [comment]
I would also greatly support the addition of being able to designate specific shared albums whose assets become part of search results and/or show up in the timeline. I just migrated from one user for my family to multiple with shared albums, without realizing that this wasn't a feature. I certainly have my regrets, but I feel like this is exactly how Immich would be implemented for this use case.
I urge for this to be added to the roadmap now that Immich is no longer in alpha, unless there's a different way to implement large amounts of shared assets without sharing your whole timeline. It's clearly a sought-after feature, and would help make Immich more usable for everyday families/users.

**Comment by createchange** [comment]
I'm sorely disappointed to see that this does not exist. I have a chat server for my family, from which I'm offloading all of the media to Immich via service account. 

I've only now just realized that no family members will be able to use the search tools over these photos. No bueno.

EDIT: I may have spoken too soon. I discovered Partner Sharing and the setting for viewing shared files in your own timeline - this permitted for the full search functionality across the images, the only thing that I'm really after here.

**Reply by DJF3** [reply]
That is a nice workaround if you want all of your "remote users" to have _full access_. And for me that is fine _for partners_ (just for the record: I only have one partner 😄 ). 
It woud be great if I gave you access to a few albums where that you would be able to use search or maps to navigate/find these photos.

**Comment by jamesdthor** [comment]
Here to also bump this for being added! Would love to have search function include shared albums! I share photos to my siblings in an album but they can't search through it! Also, it would be great if face recognition would scan through shared albums as well so they can see the faces / people on their end!
